searx/plugins/unit_converter.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Calculate mathematical expressions using ack#eval
+"""
+
+from flask_babel import gettext
+
+from searx.data import WIKIDATA_UNITS
+
+name = "Unit converter plugin"
+description = gettext("Convert between units")
+default_on = True
+
+CONVERT_KEYWORDS = ["in", "to", "as"]
+
+
+def _convert(from_value, source_si_factor, target_si_factor):
+    return from_value * source_si_factor / target_si_factor
+
+
+def _parse_text_and_convert(search, splitted_query):
+    if len(splitted_query) != 2 or splitted_query[0].strip() == "" or splitted_query[1].strip() == "":
+        return
+
+    from_value = ""
+    from_unit_key = ""
+
+    # only parse digits as value that belong together
+    read_alpha = False
+    for c in splitted_query[0]:
+        if not read_alpha and (c in ("-", ".") or str.isdigit(c)):
```

Comment:
Nitpicking: we don't deal with localization:
* `3.1415` <=> `3,1415`
* `1000km` <=> `1 000 km`

No need to do that now, the currency engine has the same issue.